### ISSUE TITLE
Rule: Wrap regex for division assignment operators

### ISSUE THREAD

**Issue by mduvall** [eslint/eslint#331]
Message: A regular expression literal can be confused with '/='

In https://github.com/nzakas/eslint/issues/324 @malandrew brought up the JSLint version of the `wrap-regex` rule.

As described there, the rule would flag for ambiguity in regular expressions that may be confused with the division assignment operator.

Example code:

``` javascript
// bad
var regex = /=1/;

// good
var x = 10;
x /= 5; // Shorthand division-assignment operator

// good
var regex = /\=1/;

// good
var regex = new RegExp("=1");
```

JSLint error: http://jslinterrors.com/a-regular-expression-literal-can-be-confused-with/
